tests/integration_test.go

```diff
@@ -826,7 +799,9 @@ func (s *IntegrationTestSuite) testSuccessConcurrentSwaps(asset types.EthAsset)
 	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
 	defer cancel()
 
-	s.setSwapTimeout(swapTimeout)
+	ac := rpcclient.NewClient(ctx, defaultXMRTakerSwapdEndpoint)
+	err := ac.SetSwapTimeout(swapTimeout)
+	require.NoError(s.T(), err)
```

Comment:
I just moved the `ac` declaration up higher. `bc` was already getting created outside of any loop and these clients are stateless from one call to the next. (If the http client has shared state, I think it was getting shared even if you created multiple clients to the same endpoint.) I removed the suite level setting of the swap timeout, so the reader can easily see that it's getting called on the taker and so that it used the shared test context.